eureka-client/src/test/java/com/netflix/discovery/util/EurekaEntityFunctionsTest.java

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.discovery.util;
+
+import com.netflix.appinfo.InstanceInfo;
+import com.netflix.discovery.shared.Application;
+import com.netflix.discovery.shared.Applications;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+
+public class EurekaEntityFunctionsTest {
+
+    private Application createSingleInstanceApp(String appId, String instanceId) {
+        InstanceInfo instanceInfo = Mockito.mock(InstanceInfo.class);
+        Mockito.when(instanceInfo.getId()).thenReturn(instanceId);
+        Mockito.when(instanceInfo.getAppName()).thenReturn(instanceId);
+        Mockito.when(instanceInfo.getStatus())
+                .thenReturn(InstanceInfo.InstanceStatus.UP);
+        Mockito.when(instanceInfo.getActionType())
+                .thenReturn(InstanceInfo.ActionType.ADDED);
+        Application application = new Application(appId);
+        application.addInstance(instanceInfo);
+        return application;
+    }
+
+    @Test
+    public void testSelectApplicationNamesIfNotNullReturnNameString() {
+        Applications applications = new Applications("appsHashCode",
+                1559658285l, new ArrayList<>(Arrays.asList(
```

Comment:
Would make sense to have a separate `createApplications` method accepting applications (throughout).

Reply:
i.e. this would be
```
Applications applications = createApplications(new Application("foo"),
    new Application("bar"), new Application("baz"));
```

Reply:
Yes I agree.. Done :)